src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/LongPollingTransportTest.java

```diff
@@ -0,0 +1,264 @@
+                        return Single.just(new HttpResponse(200, "", ""));
+                    } else if (requestCount.get() == 1) {
+                        requestCount.incrementAndGet();
+                        return Single.just(new HttpResponse(200, "", "TEST"));
+                    }
+
+                    return Single.just(new HttpResponse(204, "", ""));
+                });
+
+        Map<String, String> headers = new HashMap<>();
+        LongPollingTransport transport = new LongPollingTransport(headers, client, Single.just(""));
+
+        AtomicBoolean onReceiveCalled = new AtomicBoolean(false);
+        AtomicReference<String> message = new AtomicReference<>();
+        transport.setOnReceive((msg -> {
+            onReceiveCalled.set(true);
+            message.set(msg);
+            block.onComplete();
+        }) );
+
+        transport.setOnClose((error) -> {});
+
+        transport.start("http://example.com").timeout(1, TimeUnit.SECONDS).blockingAwait();
+        block.blockingAwait(1,TimeUnit.SECONDS);
+        assertTrue(onReceiveCalled.get());
+        assertEquals("TEST", message.get());
+    }
+
+    @Test
+    public void LongPollingTransportOnReceiveGetsCalledMultipleTimes() {
+        AtomicInteger requestCount = new AtomicInteger();
+        CompletableSubject blocker = CompletableSubject.create();
+        TestHttpClient client = new TestHttpClient()
+                .on("GET", (req) -> {
+                    if (requestCount.get() == 0) {
+                        requestCount.incrementAndGet();
+                        return Single.just(new HttpResponse(200, "", ""));
+                    } else if (requestCount.get() == 1) {
+                        requestCount.incrementAndGet();
+                        return Single.just(new HttpResponse(200, "", "FIRST"));
+                    } else if (requestCount.get() == 2) {
+                        requestCount.incrementAndGet();
+                        return Single.just(new HttpResponse(200, "", "SECOND"));
+                    }
+
+                    return Single.just(new HttpResponse(204, "", ""));
+                });
+
+        Map<String, String> headers = new HashMap<>();
+        LongPollingTransport transport = new LongPollingTransport(headers, client, Single.just(""));
+
+        AtomicBoolean onReceiveCalled = new AtomicBoolean(false);
+        AtomicReference<String> message = new AtomicReference<>("");
+        AtomicInteger messageCount = new AtomicInteger();
+        transport.setOnReceive((msg) -> {
+            onReceiveCalled.set(true);
+            message.set(message.get() + msg);
+            if (messageCount.incrementAndGet() == 2) {
+                blocker.onComplete();
+            }
+        });
+
+        transport.setOnClose((error) -> {});
+
+        transport.start("http://example.com").timeout(1, TimeUnit.SECONDS).blockingAwait();
+        blocker.blockingAwait(1, TimeUnit.SECONDS);
```

Comment:
I think if you use the timeout overload of `blockingAwait` you need to assert true on it